src/pg/mod.rs

```diff
@@ -969,27 +990,42 @@ async fn drive_future<R>(
 }
 
 fn drive_connection<S>(
-    conn: tokio_postgres::Connection<tokio_postgres::Socket, S>,
+    mut conn: tokio_postgres::Connection<tokio_postgres::Socket, S>,
 ) -> (
     broadcast::Receiver<Arc<tokio_postgres::Error>>,
+    broadcast::Receiver<diesel::pg::PgNotification>,
     oneshot::Sender<()>,
 )
 where
     S: tokio_postgres::tls::TlsStream + Unpin + Send + 'static,
 {
     let (error_tx, error_rx) = tokio::sync::broadcast::channel(1);
-    let (shutdown_tx, shutdown_rx) = tokio::sync::oneshot::channel();
+    let (notification_tx, notification_rx) = tokio::sync::broadcast::channel(1);
```

Comment:
I added the broadcast channel because it felt easier due to it being multi-receiver, but I'm not certain it's the right call because it means we could loose messages. But pondering the right approach here raises a lot of questions in my head.

1. We don't want to backpressure or hold all messages in case the user does not consume them. This means the proper treatment would be to just drop all messages not being consumed. Does this make sense?
2. What should be the behavior of calling `notification_stream` twice? Should it be `&mut` so there can only be one iterator at a time, or should we allow multiple duplicate streams? Upon writing this, I think maybe the first approach is better and more inline with sync diesel impl. Any thoughts?

Reply:
Actually, on second thought, the conn only gets notifications it has subscribed (listened) to. So maybe my worry is unwarranted and the correct implementation would be mpsc instead of broadcast.

Reply:
I think since the user asked for the `listen`, the behavior makes a lot of sense. It's not something that can blow up on it's own, it really requires misuse to behave badly.